src/app/services/sorting.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import {Application} from '../shared/application';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SortingService {
+  public propertyName: string;
+  public reverse: number;
+
+  constructor() { }
+
+  sortBy(applications: Application[], propertyName: string): Application[] {
+    if(this.propertyName === propertyName) {
+      this.reverse = -this.reverse;
+    } else {
+      this.reverse = 1;
+    }
+    this.propertyName = propertyName;
```

Comment:
This will support sorting on one property at a time only, right? Since if you sort something on one property, then sort something else on another property, then come back to the first one - you'll be back to `this.reverse = 1`?